src/credentials/LastKnownGoodTime.cpp

```diff
@@ -236,30 +196,35 @@ CHIP_ERROR LastKnownGoodTime::CommitLastKnownGoodChipEpochTime()
     return err;
 }
 
-CHIP_ERROR LastKnownGoodTime::RevertLastKnownGoodChipEpochTime()
+CHIP_ERROR LastKnownGoodTime::RevertPendingLastKnownGoodChipEpochTime()
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
-    System::Clock::Seconds32 lastKnownGoodChipEpochTime;
-    Optional<System::Clock::Seconds32> failSafeBackup;
+    System::Clock::Seconds32 storedLastKnownGoodChipEpochTime;
     VerifyOrExit(mLastKnownGoodChipEpochTime.HasValue(), err = CHIP_ERROR_INCORRECT_STATE);
-    LogTime("Last Known Good Time: ", mLastKnownGoodChipEpochTime.Value());
-    SuccessOrExit(err = LoadLastKnownGoodChipEpochTime(lastKnownGoodChipEpochTime, failSafeBackup));
-    if (!failSafeBackup.HasValue())
-    {
-        ChipLogProgress(TimeService, "No fail safe Last Known Good Time to revert to");
-        return CHIP_NO_ERROR; // if there's no value to revert to, we are done
-    }
-    LogTime("Fail safe Last Known Good Time: ", failSafeBackup.Value());
-    SuccessOrExit(err = StoreLastKnownGoodChipEpochTime(failSafeBackup.Value()));
-    mLastKnownGoodChipEpochTime.SetValue(failSafeBackup.Value());
+    LogTime("Pending Last Known Good Time: ", mLastKnownGoodChipEpochTime.Value());
+    SuccessOrExit(err = LoadLastKnownGoodChipEpochTime(storedLastKnownGoodChipEpochTime));
+    LogTime("Previous Last Known Good Time: ", storedLastKnownGoodChipEpochTime);
+    mLastKnownGoodChipEpochTime.SetValue(storedLastKnownGoodChipEpochTime);
 exit:
     if (err != CHIP_NO_ERROR)
     {
-        ChipLogError(TimeService, "Failed to persist Last Known Good Time: %" CHIP_ERROR_FORMAT, err.Format());
+        // We do not expect to arrive here unless Init() failed, the kvstore
+        // broke, or some other code removed our value from persistence.
+        // However, clearing the in-memory value is the right thing to do for
+        // all of these cases.  If there was no value known when we attempted to
+        // update to a new pending value, update will have failed and
+        // `mLastKnownGoodChipEpochTime` should already be clear, making this a
+        // no-op.  And if something else has broken, clearing our in-memory
+        // value will make it obvious that we no longer have a valid last known
+        // good time.
```

Comment:
```suggestion
        // We do not expect to arrive here unless the kvstore broke or some
        // other code removed our value from persistence.  However, clearing the
        // in-memory value is the right thing to do for all cases.  This will
        // prevent other code from consuming an invalid time.
```

Reply:
Apologies.  This is very subtle. Revert fails immediately if mLastKnownGoodChipEpochTime is already clear.  This simplified comment is more accurate.

Reply:
Applied